packages/client/src/components/agent-settings.tsx

```diff
@@ -66,21 +68,139 @@ export default function AgentSettings({ agent, agentId }: { agent: Agent; agentI
+      );
+
+      if (!confirmDelete) {
+        return;
+      }
+
+      // Set deleting state
+      setIsDeleting(true);
 
+      // Show a toast to indicate deletion is in progress
       toast({
-        title: 'Success',
-        description: 'Agent deleted successfully',
+        title: 'Deleting...',
+        description: `Deleting agent "${agent.name}"`,
       });
+
+      let responseReceived = false;
+      let navigationTimer = null;
+
+      try {
+        // Set a timeout to navigate away if the deletion takes too long
+        navigationTimer = setTimeout(() => {
+          if (!responseReceived) {
+            queryClient.invalidateQueries({ queryKey: ['agents'] });
+            navigate('/');
+            toast({
+              title: 'Note',
+              description: 'Deletion is still processing in the background.',
+            });
+          }
+        }, 8000);
+
+        const response = await apiClient.deleteAgent(agentId);
+        responseReceived = true;
+
+        if (navigationTimer) {
+          clearTimeout(navigationTimer);
+          navigationTimer = null;
+        }
+
+        // Handle partial success response
+        if (response && 'partial' in response && response.partial === true) {
+          toast({
+            title: 'Processing',
+            description: 'Deletion is still in progress and will complete in the background.',
+          });
+        } else {
+          toast({
+            title: 'Success',
+            description: 'Agent deleted successfully',
+          });
+        }
+
+        // Invalidate queries and navigate away regardless
+        queryClient.invalidateQueries({ queryKey: ['agents'] });
+        navigate('/');
+      } catch (deleteError) {
+        responseReceived = true;
+
+        if (navigationTimer) {
+          clearTimeout(navigationTimer);
+          navigationTimer = null;
+        }
+
+        // Handle specific error codes
+        if (deleteError instanceof Error) {
+          const errorMessage = deleteError.message;
+          const statusCode = (deleteError as any).statusCode;
+
+          if (
+            statusCode === 409 ||
+            errorMessage.includes('409') ||
+            errorMessage.includes('Conflict') ||
+            errorMessage.includes('foreign key constraint') ||
+            errorMessage.includes('active references')
+          ) {
+            // Conflict - agent has references that prevent deletion
+            toast({
+              title: 'Cannot Delete',
+              description:
+                'This agent cannot be deleted because it has active references. Try stopping the agent first.',
+              variant: 'destructive',
+            });
+          } else if (
+            statusCode === 408 ||
+            statusCode === 504 ||
+            errorMessage.includes('408') ||
+            errorMessage.includes('504') ||
+            errorMessage.includes('timeout') ||
+            errorMessage.includes('timed out')
```

Comment:
['408', '504', ...].includes(errorMessage) same for status code this is a cleaner way instead of doing all these conditions